habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDependenciesMojo.java

```diff
@@ -0,0 +1,156 @@
+package org.technologybrewery.habushu;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.shared.model.fileset.FileSet;
+import org.apache.maven.shared.model.fileset.util.FileSetManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Stages the source files of a monorepo dependency
+ * to the target directory along with the source files of
+ * any transitive path-based dependencies.
+ */
+
+@Mojo(name = "containerize-dependencies", defaultPhase = LifecyclePhase.VALIDATE)
+public class ContainerizeDependenciesMojo extends AbstractHabushuMojo {
+
+    private static final Logger logger = LoggerFactory.getLogger(ContainerizeDependenciesMojo.class);
+
+    @Component
+    protected MavenSession session;
+
+    protected File anchorSourceDirectory;
+    protected File anchorOutputDirectory;
+    protected final String HABUSHU = "habushu";
+
+    /**
+     * Overriding to allow execution in non-habushu projects.
+     */
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        doExecute();
+    }
+
+    @Override
+    protected void doExecute() throws MojoExecutionException, MojoFailureException {
+        if (this.anchorSourceDirectory == null) {
+            this.anchorSourceDirectory = new File(session.getExecutionRootDirectory());
+        }
+
+        Set<MavenProject> habushuProjects = getHabushuProjects();
+        try {
+            copyRelevant(habushuProjects);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Copies the relevant source files by leveraging {@link FileSet}s to filter appropriately.
+     * @param habushuProjects corresponding projects of the pom's habushu-type dependencies
+     * @throws IOException
+     */
+    private void copyRelevant(Set<MavenProject> habushuProjects) throws IOException {
+        this.anchorOutputDirectory = new File(project.getBuild().getDirectory()
+                + "/venv-support/" + this.anchorSourceDirectory.getName());
```

Comment:
A: Seems pointless to include the source directory root here as part of the output path. We could just put the files directly under `venv-support` since it's already acting as the "container" for the venv source files.  That would then allow us to move this logic to the `getAnchorOutputDirectory` method.

Reply:
This `venv-support` was nice to have, because we will eventually be adding the aggregator poetry project under it. Going to keep this for now, but will revisit this for the follow-on.